src/org/openboxprotocol/protocol/OpenBoxHeaderMatch.java

```diff
@@ -0,0 +1,113 @@
+package org.openboxprotocol.protocol;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openboxprotocol.types.Masked;
+import org.openboxprotocol.types.ValueType;
+
+public class OpenBoxHeaderMatch implements HeaderMatch {
+
+	private Map<HeaderField<?>, ValueType<?>> fields;
+	private Map<HeaderField<?>, ValueType<?>> masks;
```

Comment:
A ValueType can be an instance of the Masked class, which means that it is already masked. Thus, there is no need to have two maps here. This will prevent the chance of inconsistencies between such two maps.